src/Http/Http.Extensions/test/ValidationsGenerator/ValidationsGenerator.RecordType.cs

```diff
@@ -0,0 +1,332 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.ValidationsGenerator.Tests;
+
+public partial class ValidationsGeneratorTests : ValidationsGeneratorTestBase
+{
+    [Fact]
+    public async Task CanValidateRecordTypes()
+    {
+        // Arrange
+        var source = """
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Validation;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder();
+
+builder.Services.AddValidation();
+
+var app = builder.Build();
+
+app.MapPost("/validatable-record", (ValidatableRecord validatableRecord) => Results.Ok("Passed"!));
+
+app.Run();
+
+public class DerivedValidationAttribute : ValidationAttribute
+{
+    public override bool IsValid(object? value) => value is int number && number % 2 == 0;
+}
+
+public record SubType([Required] string RequiredProperty = "some-value", [StringLength(10)] string? StringWithLength = default);
+
+public record SubTypeWithInheritance([EmailAddress] string? EmailString, string RequiredProperty, string? StringWithLength) : SubType(RequiredProperty, StringWithLength);
+
+public static class CustomValidators
+{
+    public static ValidationResult Validate(int number, ValidationContext validationContext)
+    {
+        var parent = (ValidatableRecord)validationContext.ObjectInstance;
+        if (number == parent.IntegerWithRange)
+        {
+            return new ValidationResult(
+                "Can't use the same number value in two properties on the same class.",
+                new[] { validationContext.MemberName });
+        }
+
+        return ValidationResult.Success;
+    }
+}
+
+public record ValidatableRecord(
+    [Range(10, 100)]
```

Comment:
I thought validation attributes had to have `property:` on them to be effective -- is that not true?

I have confirmed that the attribute on the parameter _does not_ result in the range being reflected in the generated OpenApi document -- so at the very least there is some inconsistency here I think.

Reply:
> I thought validation attributes had to have property: on them to be effective -- is that not true?

That's only a constraint that exists in the MVC universe. Since we have the opportunity to deliver a better experience here I figured we'd do that.

> I have confirmed that the attribute on the parameter does not result in the range being reflected in the generated OpenApi document -- so at the very least there is some inconsistency here I think.

Yep, that's a separate bug we should file -- especially now that minimal APIs support both. Although we'll want to be mindful of only respecting it for minimal APIs. Hmmm.... 🤔